app/__tests__/api/chat/route.test.ts

```diff
@@ -1,8 +1,22 @@
 import { POST } from '../../../api/chat/route';
+jest.mock('ai', () => ({
```

Comment:
Untuk fleksibilitas lebih baik, pertimbangkan menggunakan beforeEach() agar setiap tes bisa memiliki mock yang berbeda jika dibutuhkan.
```suggestion
beforeEach(() => {
    jest.mock('ai', () => ({
        generateText: jest.fn().mockResolvedValue({
            text: 'Mocked response',
            finishReason: 'stop',
            usage: { promptTokens: 10, completionTokens: 20 },
        }),
    }));
});
```